src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -212,6 +225,299 @@ create_reference_table(PG_FUNCTION_ARGS)
 
 
 /*
+ * upgrade_reference_table accepts a distributed table which has only one shard and
+ * replicates it across all nodes to create a reference table. It also modifies related
+ * metadata to mark the table as reference explicitly.
+ */
+Datum
+upgrade_reference_table(PG_FUNCTION_ARGS)
+{
+	Oid relationId = PG_GETARG_OID(0);
+	List *shardIntervalList = NIL;
+	ShardInterval *shardInterval = NULL;
+	uint64 shardId = INVALID_SHARD_ID;
+
+	if (!IsDistributedTable(relationId))
```

Comment:
What about if a table is already `DISTRIBUTE_BY_NONE`? Should we give a useful error and exit early (i.e. `Table is already a reference table` or so)?